godpi.go

```diff
@@ -18,8 +19,39 @@ var moduleList = []types.Module{
 }
 var cacheExpiration = 5 * time.Minute
 
+// Options allow end users init module with custom options
+type Options interface {
+	Apply(types.Module)
+}
+
+type MLOption struct {
+	TCPModelPath, UDPModelPath string
+	Threshold                  float32
+}
+
+func (o MLOption) Apply(mod types.Module) {
+	lsm := mod.(*ml.LinearSVCModule)
+	if o.TCPModelPath != "" {
+		lsm.TCPModelPath = o.TCPModelPath
+	}
+	if o.UDPModelPath != "" {
+		lsm.UDPModelPath = o.UDPModelPath
+	}
+	if o.Threshold > 0.0 {
+		lsm.Threshold = o.Threshold
+	}
+}
+
 // Initialize initializes the library and the selected modules.
-func Initialize() (errs []error) {
+func Initialize(opts ...Options) (errs []error) {
+	for _, opt := range opts {
+		switch opt.(type) {
+		case MLOption:
+			opt.Apply(moduleList[2])
```

Comment:
can you improve this? looks error prone to me.

Reply:
```go
 lsm, ok := mod.(*ml.LinearSVCModule)
        if !ok {
                return
        }
```
type convert check, it that ok for you?

Reply:
Yes. use index my cause out of range panic when some guys modified the default module list or call SetModules.
The best way i can work out is by using reflect, any suggestions?

Reply:
Can't we simply call Apply on all options, with all modules? Then the option itself does what it wants with it, checking if it's been called on the appropriate module (as you currently do).

Reply:
Done.